rustls/src/crypto/mod.rs

```diff
@@ -1,12 +1,14 @@
+use crate::alias::Arc;
+
 use alloc::boxed::Box;
-use alloc::sync::Arc;
 use alloc::vec::Vec;
 use core::fmt::Debug;
 
-#[cfg(not(feature = "std"))]
+#[cfg(all(feature = "defaultproviderenabled", not(feature = "std")))]
 use once_cell::race::OnceBox;
-#[cfg(feature = "std")]
+#[cfg(all(feature = "defaultproviderenabled", feature = "std"))]
 use once_cell::sync::OnceCell;
```

Comment:
I've been thinking it may be ideal to define & use another alias for `OnceCell` vs `OnceBox`. It may be a little cleaner to break out to another PR, not sure if a separate PR is worth the trouble for such a small update for this.